chef/automate
# [infra-proxy-service] User management - GET Server users list API

### :nut_and_bolt: Description: What code changed, and why?
List the users in a server



### :chains: Related Resources
https://github.com/chef/automate/issues/5554

### :+1: Definition of Done
I have added API to list the users in a server

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs
3. Add the server admin name and admin key in the request body

Request

```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/users'```

request body
```
{
    "admin_name":"ADMIN_NAME",
    "admin_key":"ADMIN_KEY"
}
```

Response

```
{
    "users": [
       {
            "username": "sonali",
            "first_name": "Sonali",
            "last_name": "Wale",
            "email": "swale@progress.com"
        },
        {
            "username": "vinay",
            "first_name": "Vinay",
            "last_name": "Sharma",
            "email": "vsharma@chef.io"
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
